packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte

```diff
@@ -438,46 +439,51 @@ function removeExistingRegistry(registry: containerDesktopAPI.Registry) {
-            </div>
-            <div class="w-1/5">
-              <Input placeholder="Username" aria-label="Username" bind:value={newRegistryRequest.username} />
-            </div>
-            <div class="w-1/5">
-              <PasswordInput
-                id="newRegistryRequest"
-                bind:password={newRegistryRequest.secret}
-                on:action={() =>
-                  setPasswordForRegistryVisible(newRegistryRequest, !showPasswordForServerUrls.some(r => r === ''))} />
-            </div>
-            <div class="w-1/5 flex space-x-2 justify-end" role="cell">
-              <Button
-                on:click={() => loginToRegistry(newRegistryRequest)}
-                disabled={!newRegistryRequest.serverUrl || !newRegistryRequest.username || !newRegistryRequest.secret}
-                inProgress={loggingIn}>
-                Login
-              </Button>
-              <Button on:click={() => setNewRegistryFormVisible(false)} type="link">Cancel</Button>
-            </div>
-          </div>
-          <div class="flex flex-row w-full pb-3 -mt-2 pl-10">
-            <span class="font-bold whitespace-pre-line">
-              {errorResponses.find(o => o.serverUrl === newRegistryRequest.serverUrl)?.error ?? ''}
-            </span>
-          </div>
-        </div>
-        <!-- Add new registry form end -->
-      {/if}
     </div>
     <!-- Registries table end -->
   </div>
 </SettingsPage>
+
+{#if showNewRegistryForm}
+  <Dialog
+    title="Add Registry"
+    on:close={() => {
+      setNewRegistryFormVisible(false);
+    }}>
+    <div slot="content" class="flex flex-col text-[var(--pd-modal-text)] space-y-5">
+      <div>
+        <div>URL (HTTPS only)</div>
+        <Input placeholder="https://registry.io" bind:value={newRegistryRequest.serverUrl}></Input>
+      </div>
+
+      <div class="flex flex-row space-x-5 justify-stretch w-full">
+        <div class="w-full">
+          <div>Username</div>
+          <Input placeholder="username" bind:value={newRegistryRequest.username}></Input>
+        </div>
+
+        <div class="w-full">
+          <div>Password</div>
+          <PasswordInput
+            id="newRegistryRequest"
+            bind:password={newRegistryRequest.secret}
+            on:action={() =>
+              setPasswordForRegistryVisible(newRegistryRequest, !showPasswordForServerUrls.some(r => r === ''))}
+          ></PasswordInput>
+        </div>
+      </div>
+    </div>
+    <svelte:fragment slot="validation"
+      ><ErrorMessage error={errorResponses.find(o => o.serverUrl === newRegistryRequest.serverUrl)?.error ?? ''}
+      ></ErrorMessage
+      ></svelte:fragment>
+    <svelte:fragment slot="buttons">
+      <Button type="link" on:click={() => (showNewRegistryForm = false)}>Cancel</Button>
+      <Button
+        type="primary"
+        disabled={!newRegistryRequest.serverUrl || !newRegistryRequest.username || !newRegistryRequest.secret}
```

Comment:
nit:
```suggestion
        disabled={!newRegistryRequest.serverUrl.trim() || !newRegistryRequest.username.trim() || !newRegistryRequest.secret.trim()}
```

Reply:
Included in new commit (lint reformatted).